Mathlib/Algebra/Order/Group/Abs.lean

```diff
@@ -368,4 +369,32 @@ theorem eq_of_abs_sub_nonpos (h : |a - b| ≤ 0) : a = b :=
   eq_of_abs_sub_eq_zero (le_antisymm h (abs_nonneg (a - b)))
 #align eq_of_abs_sub_nonpos eq_of_abs_sub_nonpos
 
+@[simp]
+theorem abs_eq_self : |a| = a ↔ 0 ≤ a := by
+  rw [abs_eq_max_neg, max_eq_left_iff, neg_le_self_iff]
+#align abs_eq_self abs_eq_self
+
+@[simp]
+theorem abs_eq_neg_self : |a| = -a ↔ a ≤ 0 := by
+  rw [abs_eq_max_neg, max_eq_right_iff, le_neg_self_iff]
+#align abs_eq_neg_self abs_eq_neg_self
+
+/-- For an element `a` of a linear ordered ring, either `abs a = a` and `0 ≤ a`,
+    or `abs a = -a` and `a < 0`.
+    Use cases on this lemma to automate linarith in inequalities -/
+theorem abs_cases (a : α) : |a| = a ∧ 0 ≤ a ∨ |a| = -a ∧ a < 0 := by
+  by_cases h : 0 ≤ a
+  · left
+    exact ⟨abs_eq_self.mpr h, h⟩
+  · right
+    push_neg at h
+    exact ⟨abs_eq_neg_self.mpr (le_of_lt h), h⟩
+#align abs_cases abs_cases
+
+@[simp]
+theorem max_zero_add_max_neg_zero_eq_abs_self (a : α) : max a 0 + max (-a) 0 = |a| := by
```

Comment:
This lemma already exists. See `posPart` and `negPart`.

Reply:
This lemma is moved, not added. Loogle only finds [four declarations](https://loogle.lean-lang.org/?q=PosPart) about PosPart. Do you mean a version of this lemma that works for AddCommGroup already exist elsewhere?

Reply:
Your lemma is defeq to (the `symm` of) `LatticeOrderedGroup.pos_add_neg`.

Reply:
I see, thanks! Should have searched https://loogle.lean-lang.org/?q=PosPart.pos

Reply:
Unfortunately, Mathlib.Algebra.Order.LatticeGroup imports this file. The lemma is used at [two occasions](https://github.com/search?q=repo%3Aleanprover-community%2Fmathlib4%20max_zero_add_max_neg_zero_eq_abs_self&type=code) and aligned so I'm inclined to not move it for now.